### PR TITLE
amneziawg-tools: init at 1.0.20240213

### PR DESCRIPTION
**Pull request by AveryanAlex**
## Init amneziawg-tools at 1.0.20240213

Censorship-resistant VPN based on wireguard.

https://amnezia.org

Depends on #331574.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/am/amneziawg-tools/package.nix [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    "PREFIX=/"
+    "WITH_BASHCOMPLETION=yes"
+    "WITH_SYSTEMDUNITS=yes"
+    "WITH_WGQUICK=yes"
+  ];
+
+  postFixup =
+    ''
+      substituteInPlace $out/lib/systemd/system/awg-quick@.service \
+        --replace /usr/bin $out/bin
```
Comment (SuperSandro2000) [inline pkgs/by-name/am/amneziawg-tools/package.nix:47]:
```suggestion
        --replace-fail /usr/bin $out/bin
```

File: pkgs/by-name/am/amneziawg-tools/package.nix [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    + lib.optionalString stdenv.isLinux ''
+      for f in $out/bin/*; do
+        # Which firewall and resolvconf implementations to use should be determined by the
+        # environment, we provide the "default" ones as fallback.
+        wrapProgram $f \
+          --prefix PATH : ${
+            lib.makeBinPath [
+              procps
+              iproute2
+            ]
```
Comment (paveloom) [inline pkgs/by-name/am/amneziawg-tools/package.nix:58]: When starting the `systemd` unit, `awg-quick` complained about `readlink` missing. I added `coreutils` here to resolve it.

Comment (averyanalex) [reply pkgs/by-name/am/amneziawg-tools/package.nix:58]: I checked awg-quick@name service, it doen't work, like the wg-quick@name. In [wg-quick module](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/wg-quick.nix) some tools, like `config.networking.firewall.package` are added to service path. I think best solution is to update wireguard modules and add something like `type = wireguard | amneziawg` option.

**Comment by cab404** [comment]
@averyanalex anything left there you want to do before merging?

**Comment by averyanalex** [comment]
> @averyanalex anything left there you want to do before merging?

New version just released (18.10.2024) with minor improvements, I will update.

Also, what do you think about https://github.com/NixOS/nixpkgs/pull/331582#pullrequestreview-2301345095? Should "vanilla" systemd services work out of the box?

**Comment by cab404** [comment]
> > @averyanalex anything left there you want to do before merging?
> 
> New version just released (18.10.2024) with minor improvements, I will update.
> 
> Also, what do you think about [#331582 (review)](https://github.com/NixOS/nixpkgs/pull/331582#pullrequestreview-2301345095)? Should "vanilla" systemd services work out of the box?

It would be nice if they did. Not everybody have fully managed wg/awg configs.

**Comment by averyanalex** [comment]
> > > @averyanalex anything left there you want to do before merging?
> > 
> > 
> > New version just released (18.10.2024) with minor improvements, I will update.
> > Also, what do you think about [#331582 (review)](https://github.com/NixOS/nixpkgs/pull/331582#pullrequestreview-2301345095)? Should "vanilla" systemd services work out of the box?
> 
> It would be nice if they did. Not everybody have fully nixos managed wg/awg configs.

If I add coreutils to PATH, systemd service starts without errors, but uses iptables on nftables system and doesn't work. I think this can be merged as is.

**Comment by cab404** [comment]
> > > > @averyanalex anything left there you want to do before merging?
> > > 
> > > 
> > > New version just released (18.10.2024) with minor improvements, I will update.
> > > Also, what do you think about [#331582 (review)](https://github.com/NixOS/nixpkgs/pull/331582#pullrequestreview-2301345095)? Should "vanilla" systemd services work out of the box?
> > 
> > 
> > It would be nice if they did. Not everybody have fully nixos managed wg/awg configs.
> 
> If I add coreutils to PATH, systemd service starts without errors, but uses iptables on nftables system and doesn't work. I think this can be merged as is.

Nice, thank you!

**Pull request merged**